internal/constellation/helm/generateCertManager.sh

```diff
@@ -12,10 +12,51 @@ function cleanup {
 
 trap cleanup EXIT
 
+version="1.12.6"
 helm pull cert-manager \
-  --version 1.12.6 \
+  --version ${version} \
   --repo "https://charts.jetstack.io" \
   --untar \
   --untardir "charts"
 
+
+get_sha256_hash() {
+  local component="$1"
+  local url="https://quay.io/v2/jetstack/${component}/manifests/v${version}"
+  local hash
+  set -e  # Exit immediately if a command fails
+  has_error=$(curl -s -L -H "Accept: application/vnd.docker.distribution.manifest.v2+json" "$url" | jq 'has("errors")')
+
+
+  # Check if 'errors' attribute exists
+  if [ "$has_error" = "true" ]; then
+      echo "Errors attribute found. Exiting with status 1."
+      exit 1
+  fi
+
+  hash=$(curl -s -L -H "Accept: application/vnd.docker.distribution.manifest.v2+json" "$url" | sha256sum | awk '{print $1}')
+
+  set +e  # Disable the 'set -e' option to avoid exiting on subsequent commands
+
+  echo "$hash"
```

Comment:
`set -e` is inherited from line 3, and `set +e` resets globally. I don't think you need to change this at all.

```suggestion
  has_error=$(curl -s -L -H "Accept: application/vnd.docker.distribution.manifest.v2+json" "$url" | jq 'has("errors")')

  # Check if 'errors' attribute exists
  if [ "$has_error" = "true" ]; then
      echo "Errors attribute found. Exiting with status 1."
      exit 1
  fi

  curl -s -L -H "Accept: application/vnd.docker.distribution.manifest.v2+json" "$url" | sha256sum | awk '{print $1}'
```